pkg/agent/utils.go

```diff
@@ -586,3 +586,25 @@ func addFeatureGateString(featureGates string, key string, value bool) string {
 	}
 	return strings.Join(pairs, ",")
 }
+
+// ParseCSE parses the raw VMSS CSE output
+func ParseCSE(message string) (datamodel.VMSSInstanceViewCSEStatus, error) {
+	var cseStatus datamodel.VMSSInstanceViewCSEStatus
+	var rerr error
+	start := strings.Index(message, "[stdout]") + len("[stdout]")
+	end := strings.Index(message, "[stderr]")
+	if end > start {
+		rawInstanceViewInfo := message[start:end]
+		err := json.Unmarshal([]byte(rawInstanceViewInfo), &cseStatus)
+		if err != nil || cseStatus.ExitCode == "" {
+			// Regex "vmssInstanceErrorCode=" is part of contract to parse the error, please inform clients who are relying on it before change the regex.
```

Comment:
actually this func looks very vmss specific, are we sure to make it generic?

Reply:
If we are to make it generic, maybe ParseCSEMessage(message, type) and type can be VMSS or VMAS. And depending on the type, it may generate different error

Reply:
yes, if we want to make this generic, we should have the method changed to ParseCSE(message, type string). And the error in this method should be like fmt.Errorf("%sCSE has invalid message=%s, %s=%s", type, message, InstanceErrorCode, InvalidCSEMessage).
CSE message format for both VMAS and VMSS are the same? We can still use the same method to parse?

Reply:
it should be generic. The error message from this method should contain NO vmss or vmas keywords. The client can interpret that.

Reply:
sounds like this func should be just used to parse cse message -> json. VMSS/VMAS logic will be handled in RP